gwctl/pkg/printer/policies_test.go

```diff
@@ -354,3 +354,155 @@ timeoutpolicies.bar.com      Direct       Namespaced  5m
+				Names: apiextensionsv1.CustomResourceDefinitionNames{
+					Plural: "healthcheckpolicies",
+					Kind:   "HealthCheckPolicy",
+				},
+			},
+		},
+		&unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "foo.com/v1",
+				"kind":       "HealthCheckPolicy",
+				"metadata": map[string]interface{}{
+					"name": "health-check-gateway",
+				},
+				"spec": map[string]interface{}{
+					"override": map[string]interface{}{
+						"key1": "value-child-1",
+					},
+					"default": map[string]interface{}{
+						"key2": "value-child-2",
+						"key5": "value-child-5",
+					},
+					"targetRef": map[string]interface{}{
+						"group":     "gateway.networking.k8s.io",
+						"kind":      "Gateway",
+						"name":      "foo-gateway",
+						"namespace": "default",
+					},
+				},
+			},
+		},
+
+		&apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "timeoutpolicies.bar.com",
+				Labels: map[string]string{
+					gatewayv1alpha2.PolicyLabelKey: "direct",
+				},
+			},
+			Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+				Scope:    apiextensionsv1.NamespaceScoped,
+				Group:    "bar.com",
+				Versions: []apiextensionsv1.CustomResourceDefinitionVersion{{Name: "v1"}},
+				Names: apiextensionsv1.CustomResourceDefinitionNames{
+					Plural: "timeoutpolicies",
+					Kind:   "TimeoutPolicy",
+				},
+			},
+		},
+		&unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "bar.com/v1",
+				"kind":       "TimeoutPolicy",
+				"metadata": map[string]interface{}{
+					"name": "timeout-policy-namespace",
+				},
+				"spec": map[string]interface{}{
+					"condition": "path=/abc",
+					"seconds":   int64(30),
+					"targetRef": map[string]interface{}{
+						"kind": "Namespace",
+						"name": "default",
+					},
+				},
+			},
+		},
+	}
+
+	params := utils.MustParamsForTest(t, common.MustClientsForTest(t, objects...))
+	pp := &PoliciesPrinter{
+		Out:   &bytes.Buffer{},
+		// Clock: fakeClock,
+	}
+	pp.PolicyCrd_PrintDescribeView(params.PolicyManager.GetCRDs())
+
+	got := pp.Out.(*bytes.Buffer).String()
+	want := `
+Name: healthcheckpolicies.foo.com
+Labels:
+  gateway.networking.k8s.io/policy: inherited
+APIVersion: apiextensions.k8s.io/v1
+Kind: CustomResourceDefinition
+Metadata:
+  creationTimestamp: null
+  labels:
+    gateway.networking.k8s.io/policy: inherited
+  name: healthcheckpolicies.foo.com
+  resourceVersion: "999"
+Spec:
+  group: foo.com
+  names:
+    kind: HealthCheckPolicy
+    plural: healthcheckpolicies
+  scope: Cluster
+  versions:
+  - name: v1
+    served: false
+    storage: false
+Status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: null
+  storedVersions: null
+
+
+Name: timeoutpolicies.bar.com
+Labels:
+  gateway.networking.k8s.io/policy: direct
+APIVersion: apiextensions.k8s.io/v1
+Kind: CustomResourceDefinition
+Metadata:
+  creationTimestamp: null
+  labels:
```

Comment:
For things like `label`, `annotations` etc which we're including outside the `metadata`, maybe we should not print them twice?

Reply:
Agreed, I will remove the `label` and `annotations` outside as its more difficult to remove from within `metadata`

Reply:
Lets try using concrete types and see if that helps ease this as well.